src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStore.cs

```diff
@@ -298,6 +299,29 @@ internal class EnumStore
 
         protected virtual IDictionary<string, string> EnumDict => _enums;
 
+        internal EnumStore WithRequiredEnums(IEnumerable<TexlFunction> functions)
+        {
+            var missingEnums = new Dictionary<string, string>();
+            foreach (var function in functions)
+            {
+                var requiredEnumNames = function.GetRequiredEnumNames();
+                foreach (var name in requiredEnumNames)
+                {
+                    if (!EnumDict.ContainsKey(name) && !missingEnums.ContainsKey(name))
+                    {
+                        missingEnums.Add(name, _enums[name]);
```

Comment:
>name [](http://example.com/codeflow?start=41&length=4)

is this possible to have collisions?

Reply:
If the user added their own enum with the same name as one of the enums that are required by the functions, then the required enum will not get added.